### PR TITLE
Move some compilation logic from the executable to compile.dart

### PR DESCRIPTION
**Pull request by nex3**
This doesn't reduce code size, but it does make it easier to add
functionality to *all* Sass compilations.

### CODE REVIEW

File: lib/src/async_compile.dart [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+/// * `importers`, `packageResolver`, and/or `loadPaths`.
+Future<CompileResult> compileAsync(String path,
+    {Syntax syntax,
+    Logger logger,
+    AsyncImportCache importCache,
+    NodeImporter nodeImporter,
+    Iterable<AsyncImporter> importers,
+    Iterable<String> loadPaths,
+    SyncPackageResolver packageResolver,
+    Iterable<AsyncCallable> functions,
```
[the remark this answers is not in the document]
Comment (nex3) [reply lib/src/async_compile.dart]: Done.

**Comment by nshahan** [comment]
Are these Travis failures expected?

**Comment by nex3** [comment]
Nope!

**Comment by nex3** [comment]
@nshahan They're fixed now.

**Pull request merged**